### PR TITLE
Add PFFT parallel FFT package

### PR DESCRIPTION
**Pull request by mkuron**
This pull request adds a package for the PFFT parallel FFT library, which is based on FFTW and provides MPI-parallel multi-dimensional FFTs, while FFTW itself only implements single-dimensional FFTs in parallel.

Unfortunately, it currently requires a patch to be applied to FFTW. This pull request therefore includes changes to add an extra variant to the FFTW package.

For details, see the developer's website, https://www-user.tu-chemnitz.de/~potts/workgroup/pippig/software.php.

### CODE REVIEW

File: var/spack/repos/builtin/packages/pfft/package.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        if not self.compiler.f77 or not self.compiler.fc:
+            options.append("--disable-fortran")
+
+        configure(*options)
+        make()
+        if self.run_tests:
+            make("check")
+        make("install")
+
+        if '+float' in spec:
```
Comment (davydden) [inline var/spack/repos/builtin/packages/pfft/package.py]: not sure what is the correct way to check variants in `fftw`. Did you check that this works? I would say it does not.

@tgamblin @alalazo :  do you know how to check variants of a dependency in the dependent package?

Comment (tgamblin) [inline var/spack/repos/builtin/packages/pfft/package.py]: I would write: `if '+float' in spec['fftw']`

**Comment by tgamblin** [comment]
If the patch is publicly available, can you add it as a URL with a checksum, as @alalazo did [in nwchem](https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/nwchem/package.py)?

File: var/spack/repos/builtin/packages/pfft/package.py (hunk 1 above)
Comment (mkuron) [inline var/spack/repos/builtin/packages/pfft/package.py]: Changed that, thanks for noticing. I had simply copied that from the FFTW package and forgot to adapt it.

**Comment by mkuron** [comment]
I don't think the patch is available for download anywhere, unfortunately. The developer only provides a shell script that patches and compiles FFTW 3.3.4. While I could automatically extract the patch from that script, that would only work for FFTW 3.3.4 and not be a very elegant solution. I ended up getting the 3.3.5 patch from MacPorts (https://github.com/macports/macports-ports/commits/master/math/fftw-3/files/patch-pfft.diff) because they had already adapted it.

I've taken care of the dependency variants problem.

**Comment by healther** [comment]
Since doing a fresh pull from the github repo I have trouble installing py-pyfftw

```
spack install py-pyfftw
==> Error: No such patch file for package fftw: /pfs/data2/home/hd/hd_hd/hd_wv385/git/_spack_repos/spack/var/spack/repos/builtin/packages/fftw/pfft-3.3.5.patch
```

Is this related to this PR? If so how do I fix this for my local clone? I cannot find either commit hashes ( 6f2c05b or c655049c) in my local clone..
I'm sure I'm just missing something really obvious and being totally blind about it... Any help would be appreciated.

**Comment by davydden** [comment]
@healther : did you check if the `pfft-3.3.5.patch` exists locally in `...../packages/fftw/`?

**Comment by healther** [comment]
Ah yeah I found the problem... I had added a builtin package repo from a forked spack version, which itself was not up to date and hence shadowed the builtin from my local repository...

Is there a way to give a search preference in which of the repo's spack should look first? ```repo add``` seems to be able to only prepend new repositories.

**Pull request merged**